ckan/lib/helpers.py

```diff
@@ -725,28 +726,46 @@ def get_facet_items_dict(facet, limit=None, exclude_active=False):
     facet -- the name of the facet to filter.
     limit -- the max. number of facet items to return.
     exclude_active -- only return unselected facets.
-
+    int_sort -- sort facets with int literals in 'name' by name and descending
     '''
-    if not c.search_facets or \
-            not c.search_facets.get(facet) or \
-            not c.search_facets.get(facet).get('items'):
+
+    def isint(v):
+        try:
+            i = int(v)
+        except:
+            i = v
+        return isinstance(i, int)
```

Comment:
You don't need a separate call to `isinstance`: You can return `True` if `int(v)` doesn't raise an exception and `False` otherwise.